controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -219,6 +171,11 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 			r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
 		}
 
+		cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+		if err := r.Client.List(r.ctx, cephBlockPoolList); err != nil && len(cephBlockPoolList.Items) > 0 {
+			r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
+		}
```

Comment:
This condition is not enough, you are searching for any ceph block pool, not the ones that are tied to this specific consumer. This means that if a single consumer has a block pool provisioned for it all consumers will pass this test - this behavior is a bug.

To solve the issue, you need to add a filter option based on the consumer name label to your list operation.